packages/web-components/src/components/ic-dialog/ic-dialog.tsx

```diff
@@ -0,0 +1,403 @@
+      const alert = this.el.shadowRoot.querySelector("ic-alert");
+      alert.setAttribute("variant", this.status);
+    }
+  };
+
+  private getInteractiveElements = () => {
+    this.interactiveElementList = Array.from(
+      this.el.shadowRoot.querySelectorAll("ic-button")
+    );
+    const slottedInteractiveElements = Array.from(
+      this.el.querySelectorAll(
+        `a[href], button, input, textarea, select, details, [tabindex]:not([tabindex="-1"]), 
+        ic-button, ic-checkbox, ic-select, ic-search-bar, ic-tab-group, ic-radio-group, 
+        ic-back-to-top, ic-breadcrumb, ic-chip[dismissible="true"], ic-footer-links, ic-link, ic-navigation-button, 
+        ic-navigation-item, ic-switch`
+      )
+    );
+    if (slottedInteractiveElements.length > 0) {
+      if (slottedInteractiveElements[0].slot !== "dialog-controls") {
+        slottedInteractiveElements[0].setAttribute("data-gets-focus", "");
+      } else if (!this.destructive) {
+        slottedInteractiveElements[
+          slottedInteractiveElements.length - 1
+        ].setAttribute("data-gets-focus", "");
+      }
+    }
+    for (let i = 0; i < slottedInteractiveElements.length; i++) {
+      this.interactiveElementList.splice(
+        1 + i,
+        0,
+        slottedInteractiveElements[i] as HTMLElement
+      );
+    }
+  };
+
+  private focusNextInteractiveElement = (shiftKey: boolean) => {
+    if (shiftKey) {
+      this.focusedElementIndex -= 1;
+    } else {
+      this.focusedElementIndex += 1;
+    }
+
+    if (this.focusedElementIndex > this.interactiveElementList.length - 1)
+      this.focusedElementIndex = 0;
+    else if (this.focusedElementIndex < 0) {
+      this.focusedElementIndex = this.interactiveElementList.length - 1;
+    }
+
+    if (process.env.JEST_WORKER_ID === undefined) {
+      (
+        this.interactiveElementList[this.focusedElementIndex] as HTMLElement
+      ).focus();
+    }
+  };
+
+  private setButtonOnclick = () => {
+    if (this.buttons) {
+      this.buttonOnclick0 = new Function(this.buttonProps[0]?.onclick);
+      this.buttonOnclick1 = new Function(this.buttonProps[1]?.onclick);
+      this.buttonOnclick2 = new Function(this.buttonProps[2]?.onclick);
+    }
+  };
+
+  private getButtonOnclick = (index: number) => {
+    if (index === 0) {
+      return this.buttonOnclick0();
+    } else if (index === 1) {
+      return this.buttonOnclick1();
+    } else {
+      return this.buttonOnclick2();
+    }
+  };
+
+  private getButtonVariant = (index: number) => {
+    if (this.buttonProps.length === 1) {
+      return this.destructive ? "destructive" : "primary";
+    } else if (this.buttonProps.length === 2) {
+      if (index === 0) {
+        return "tertiary";
+      } else {
+        return this.destructive ? "destructive" : "primary";
+      }
+    } else {
+      if (index === 2) {
+        return this.destructive ? "destructive" : "primary";
+      } else {
+        return "secondary";
+      }
+    }
+  };
```

Comment:
Wonder if you could create a variable to avoid repeating the `this.destructive` check, e.g. something like this (probably not completely necessary but up to you):
```suggestion
  private getButtonVariant = (index: number) => {
    const mainVariant = this.destructive ? "destructive" : "primary";
    if (this.buttonProps.length === 1) {
      return mainVariant;
    } else if (this.buttonProps.length === 2) {
      if (index === 0) {
        return "tertiary";
      } else {
        return mainVariant;
      }
    } else {
      if (index === 2) {
        return mainVariant;
      } else {
        return "secondary";
      }
    }
  };
```

Reply:
Good idea, I have added this